### PR TITLE
Cake 2100 Adding mobile global nav to custom element

### PR DESCRIPTION
**Pull request by lizlux**
https://wikia-inc.atlassian.net/browse/CAKE-2100
@Wikia/cake 

This is a huge PR with lot's of refactoring including: 

* De-componentizing the global nav element for desktop so it's just an ES6 class
* Adding a class for handling the mobile version
** Includes copying HTML and CSS from Mobile Wiki
* Creating a parent element that instantiates both desktop and mobile DOMs
* Moving event handling and state management to the parent element
* Refactoring the attributes helper - though we may want to revert some of this as it wasn't super necessary in the end
* Unifying hiding search on desktop and mobile

Some improvements that would be nice: 
* Making attribute handling more developer friendly (i.e. camelCase instead of dashed-case to access attributes) _- done - mostly reverted my changes_
* General code cleanup, giving some more thought to the directory structure
* Consider creating a model for accessing MW API response data

### CODE REVIEW

File: src/components/footer/FandomGlobalFooter.es6 [hunk 1]
```diff
@@ -12,11 +12,11 @@ import designSystemStyle from 'design-system/dist/css/styles.css';
 export default class FandomGlobalFooter extends HTMLElement {
     connectedCallback() {
         this.rootElement = this.attachShadow({ mode: 'open' });
-        this.atts = new AttributeHelper(this);
+        this.atts = new AttributeHelper(this).getAllAttributes();
         this.svgs = new SvgHelper(this.rootElement);
-        this.strings = getStrings(this.atts.langCode);
+        this.strings = getStrings(this.atts['lang-code']);
```
Comment (nmonterroso) [inline src/components/footer/FandomGlobalFooter.es6]: can we create constants out of these string literals?

File: src/components/header/FandomGlobalHeaderDesktop.es6 [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+        this._updateNavLinks();
+
+        this._bindSearchActions();
+
+        this.parent.onEvent(EVENTS.USER_DATA_REFRESHED, () => this._updateUserState());
+
+        return this;
+    }
+
+    isVisible() {
```
Comment (nmonterroso) [inline src/components/header/FandomGlobalHeaderDesktop.es6]: we call this from the FC in `StickyController`. If we're going to remove it from here we need to make sure to account for the height diff somehow (maybe have it listen to the pin/unpinned event?)

**Review by nmonterroso — COMMENTED** [review]
The isolation of the events looks awesome. Some notes/issues:

- In desktop view (all browsers) the top of the content is cut off:
image
- in desktop firefox the component is slightly lower than the top of the page:
image
- when making a narrow window in safari the view doesn't switch over:
image
- the mobile view styles could use `cursor: pointer;` for the signin/register links
- when in mobile view the desktop event for headroom is still being fired. Maybe the parent needs to have a notion of which of its view is "active" and the children should somehow be able to enable/disable?
- the mobile view doesn't look like it's firing events for link clicks (logo, vertical links)
- unused imports in the child classes
- clicking the menu expand in mobile view adds `#` to the url bar

Let me know how I can help resolve these.

**Comment by nmonterroso** [comment]
With the latest changes pushed up I think at this point the remaining issues are:
- when making a narrow window in safari the view doesn't switch over
- when in mobile view the desktop event for headroom is still being fired. Maybe the parent needs to have a notion of which of its view is "active" and the children should somehow be able to enable/disable?
- comment on `isVisible`: we call this from the FC in StickyController. If we're going to remove it from here we need to make sure to account for the height diff somehow (maybe have it listen to the pin/unpinned event?)

**Comment by lizlux** [comment]
- [x] safari responsiveness
- [x] disable headroom in mobile
- [x] `isVisible` on custom element

**Comment by nmonterroso** [comment]
LGTM

**Comment by lizlux** [comment]
Oops, meant to tag @vforge on this too.

**Pull request merged**